src/test/java/com/redhat/cloud/notifications/db/BehaviorGroupResourcesTest.java

```diff
@@ -349,19 +326,24 @@ private Boolean muteEventType(String accountId, UUID eventTypeId) {
         return behaviorGroupResources.findBehaviorGroupsByEventTypeId(accountId, eventTypeId, new Query()).await().indefinitely();
     }
 
-    private Boolean addBehaviorGroupAction(String accountId, UUID behaviorGroupId, UUID endpointId) {
-        return behaviorGroupResources.addBehaviorGroupAction(accountId, behaviorGroupId, endpointId).await().indefinitely();
-    }
-
-    private List<BehaviorGroupAction> findBehaviorGroupActionsByBehaviorGroupId(UUID behaviorGroupId) {
-        String query = "FROM BehaviorGroupAction WHERE behaviorGroup.id = :behaviorGroupId";
-        return session.createQuery(query, BehaviorGroupAction.class)
-                .setParameter("behaviorGroupId", behaviorGroupId)
-                .getResultList()
-                .await().indefinitely();
+    private void updateAndCheckBehaviorGroupActions(String accountId, UUID bundleId, UUID behaviorGroupId, boolean expectedResult, UUID... endpointIds) {
+        Boolean updated = behaviorGroupResources.updateBehaviorGroupActions(accountId, behaviorGroupId, Arrays.asList(endpointIds)).await().indefinitely();
+        // Is the update result the one we expected?
+        assertEquals(expectedResult, updated);
+        if (expectedResult) {
+            session.clear(); // We need to clear the session L1 cache before checking the update result.
```

Comment:
This looks like something that keeps appearing - At least is the second time we have a problem with the cache IIRC.
In this case, is it because we are using a single session for the whole suite and it doesn't "pick" the changes?

Reply:
Yes, that's exactly the reason. Here's what's happening:
- First we `persist` or `select` an entity which is usually (not always) cached into the L1 cache.
- Then we update it with an HQL or a native query which does not affect the L1 cache content.
- Finally we select the entity again and Hibernate returns the L1 cache entity state because it's not aware that the entity was updated in the database with the HQL/native query.

It only happens in tests where `*Resources` are injected and used to execute multiple queries. It can be a bit annoying while writing tests, but I don't think it's a problem in the end. Testing all database operations directly (not through the app REST APIs) is important for the tests coverage and reliability. Clearing the Hibernate session is a small price to pay to reach this goal 😃